### PR TITLE
add 3. Longest Substring Without Repeating Characters

### PR DESCRIPTION
**Pull request by kobakobashu**
time : 35m
Please check my code.

### CODE REVIEW

File: 3_longest-substring-without-repeating-characters.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        """
+        time: O(len(s))
+        space: O(len(s))
+        """
+        ref = {}
+        cur_max_length = 0
+        start = 0
+        for idx, alph in enumerate(s):
+            if alph in ref:
+                sums = ref[alph] + 1
```
Comment (ryuji0123) [inline 3_longest-substring-without-repeating-characters.py:19]: sums is unclear

**Review by ryuji0123 — CHANGES_REQUESTED** [review]

**Comment by kobakobashu** [comment]
I changed the unclear name "sums", and fixed the time complexity.  Please check my code.

**Comment by kobakobashu** [comment]
I fixed the time complexity again.  Please check my code.

**Review by tkkawa — APPROVED** [review]
LGTM

**Pull request closed without merging**